### PR TITLE
[Merged by Bors] - chore(install_macos.sh): Don't overwrite existing vscode

### PR DESCRIPTION
**Pull request by JasonGross**
If this script is run without sudo and VSCode is already installed, it seems to break VSCode.  There's no need to install it VSCode if it's already installed, in any case.

(We could do the same with elan, I'm less sure about the tradeoffs there.)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

Supersedes #18263

### CODE REVIEW

**Review by YaelDillies — APPROVED** [review]
Looks easy enough. Thanks!

maintainer merge

File: scripts/install_macos.sh [hunk 1]
```diff
@@ -14,17 +14,22 @@ curl https://raw.githubusercontent.com/leanprover/elan/master/elan-init.sh -sSf
...
 
-# Unzip the downloaded file to the Applications folder
-unzip -o ~/Downloads/VSCode-darwin-universal.zip -d /Applications
+    # Unzip the downloaded file to the Applications folder
+    unzip -o ~/Downloads/VSCode-darwin-universal.zip -d /Applications
 
-# Add the VS Code binary to the PATH to enable launching from the terminal
-cat << EOF >> ~/.zprofile
+  # Add the VS Code binary to the PATH to enable launching from the terminal
+  cat >> ~/.zprofile << EOF
```
Comment (j-loreaux) [inline scripts/install_macos.sh:26]: to match the indentation of the `if` block
```suggestion
    # Add the VS Code binary to the PATH to enable launching from the terminal
    cat >> ~/.zprofile << EOF
```

**Review by j-loreaux — COMMENTED** [review]
Thanks!

bors d+

**Pull request closed without merging**